examples/vmseries/main.tf

```diff
@@ -0,0 +1,339 @@
+  resource_group_name          = local.resource_group.name
+  location                     = var.location
+  platform_update_domain_count = each.value.update_domain_count
+  platform_fault_domain_count  = each.value.fault_domain_count
+
+  tags = var.tags
+}
+
+module "vmseries" {
+  source = "../../modules/vmseries"
+
+  for_each = var.vmseries
+  # for_each = {}
+
+  name                = "${var.name_prefix}${each.value.name}"
+  location            = var.location
+  resource_group_name = local.resource_group.name
+
+  authentication = local.authentication[each.key]
+  image          = each.value.image
+  virtual_machine = merge(
+    each.value.virtual_machine,
+    {
+      disk_name = "${var.name_prefix}${coalesce(each.value.virtual_machine.disk_name, "${each.value.name}-osdisk")}"
+      avset_id  = try(azurerm_availability_set.this[each.value.virtual_machine.avset_key].id, null)
+      bootstrap_options = try(
+        coalesce(
+          each.value.virtual_machine.bootstrap_options,
+          join(",", [
+            "storage-account=${module.bootstrap[each.value.virtual_machine.bootstrap_package.bootstrap_storage_key].storage_account_name}",
+            "access-key=${module.bootstrap[each.value.virtual_machine.bootstrap_package.bootstrap_storage_key].storage_account_primary_access_key}",
+            "file-share=${each.key}",
+            "share-directory=None"
+          ]),
+        ),
+        null
+      )
+    }
+  )
+
+  interfaces = [for v in each.value.interfaces : {
+    name                     = "${var.name_prefix}${v.name}"
+    subnet_id                = module.vnet[each.value.virtual_machine.vnet_key].subnet_ids[v.subnet_key]
+    create_public_ip         = v.create_public_ip
+    public_ip_name           = v.create_public_ip ? "${var.name_prefix}${coalesce(v.public_ip_name, "${each.value.name}-pip")}" : v.public_ip_name
+    public_ip_resource_group = v.public_ip_resource_group
+    private_ip_address       = v.private_ip_address
+  }]
+
+  tags = var.tags
+  depends_on = [
+    module.vnet,
+    azurerm_availability_set.this,
+    module.load_balancer
+    # module.bootstrap,
+    # module.bootstrap_share
+  ]
+}
+
+
+locals {
+  full_nics_list_flat = flatten([
+    for vm_key, vm in var.vmseries : [
+      for nic in vm.interfaces : {
+        vm       = vm_key
+        nic_name = "${var.name_prefix}${nic.name}"
+        lb_key   = nic.load_balancer_key
+      } if nic.load_balancer_key != null
+    ]
+  ])
+  backend_pool_ids = { for v in local.full_nics_list_flat : "${v.vm}-${v.nic_name}" => v }
+}
+
+resource "azurerm_network_interface_backend_address_pool_association" "this" {
+  for_each = local.backend_pool_ids
+
+  backend_address_pool_id = module.load_balancer[each.value.lb_key].backend_pool_id
+  ip_configuration_name   = module.vmseries[each.value.vm].interfaces[each.value.nic_name].ip_configuration[0].name
+  network_interface_id    = module.vmseries[each.value.vm].interfaces[each.value.nic_name].id
+}
```

Comment:
Still not sure about adding this resource into the glue code, hope we can discuss on call.